velox/sdk/cpp/expression/ExpressionUtils.hpp

```diff
@@ -0,0 +1,236 @@
+    std::vector<std::string> jsons =
+        ConvertJStringArrayToVector(env, exprJsons);
+    for (const auto& json : jsons) {
+      typeExprs.push_back(
+          T::deserialize(folly::parseJson(json, getSerializationOptions())));
+    }
+    return typeExprs;
+  }
+
+  static exec::ExprSet compileExpression(
+      const std::string& text,
+      const TypePtr& rowType) {
+    parse::ParseOptions options_;
+    auto previousHook = core::Expressions::getResolverHook();
+    parse::registerTypeResolver();
+    auto untyped = parse::parseExpr(text, options_);
+    auto typed = core::Expressions::inferTypes(
+        untyped,
+        rowType,
+        sdk::memory::MemoryManager::get()->dictExecutionMemoryPool().get());
+    core::Expressions::setTypeResolverHook(previousHook);
+    std::shared_ptr<core::QueryCtx> queryCtx_ = core::QueryCtx::create();
+    core::ExecCtx execCtx_{
+        sdk::memory::MemoryManager::get()->vectorBatchMemoryPool().get(),
+        queryCtx_.get()};
+    return exec::ExprSet({typed}, &execCtx_);
+  }
+
+  static VectorPtr evaluate(
+      exec::ExprSet& exprSet,
+      const RowVectorPtr& data,
+      const SelectivityVector& rows) {
+    std::shared_ptr<core::QueryCtx> queryCtx_ = core::QueryCtx::create();
+    core::ExecCtx execCtx_{
+        sdk::memory::MemoryManager::get()->dictExecutionMemoryPool().get(),
+        queryCtx_.get()};
+    exec::EvalCtx evalCtx(&execCtx_, &exprSet, data.get());
+    std::vector<VectorPtr> results(1);
+    exprSet.eval(rows, evalCtx, results);
+    return results[0];
+  }
+
+  static VectorPtr evaluate(exec::ExprSet& exprSet, const RowVectorPtr& data) {
+    SelectivityVector rows(data->size());
+    return evaluate(exprSet, data, rows);
+  }
+
+  static VectorPtr evaluate(
+      const std::string& expression,
+      const RowVectorPtr& data) {
+    auto exprSet = compileExpression(expression, asRowType(data->type()));
+    return evaluate(exprSet, data);
+  }
+
+  static TypePtr jsonToVeloxType(std::string json) {
+    return convertSparkStructToVelox(
+        folly::parseJson(json, getSerializationOptions()));
+  }
+
+  // 将Spark的字段类型转换为Velox的Type
+  static TypePtr convertSparkFieldToVelox(const folly::dynamic& field) {
+    if (field["type"].isObject()) {
+      return convertSparkStructToVelox(field);
+    }
+    auto type = field["type"].asString();
+
+    // 这个例子没有处理nullable字段，实际使用时你可能需要考虑这个字段
+
+    if (type == "integer") {
+      return INTEGER();
+    } else if (type == "long") {
+      return BIGINT();
+    } else if (type == "string") {
+      return VARCHAR();
+    } else if (type == "double") {
+      return DOUBLE();
+    } else if (type == "date") {
+      return DATE();
+    } else if (type == "timestamp") {
+      return TIMESTAMP();
+    } else {
+      throw std::runtime_error("Unsupported type: " + type);
+    }
+  }
+
+  // 将Spark的StructType转换为Velox的Type
+  static TypePtr convertSparkStructToVelox(const folly::dynamic& sparkStruct) {
```

Comment:
I was expecting this binding layer to closely follow velox's API. Why are there Spark-specific conversion code? Can we get rid of it?

Reply:
Currently, I am temporarily passing the Spark schema catalog json to C++, otherwise I would need to implement it myself. Is there a need to make a change here?

Reply:
done